x-pack/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -247,6 +248,12 @@ export function XYChart({
     ({ id }) => id === filteredLayers[0].xAccessor
   );
   const xAxisFormatter = formatFactory(xAxisColumn && xAxisColumn.formatHint);
+  const layersAlreadyFormatted: Record<string, boolean> = {};
+  // This is a safe formatter for the xAccessor that abstracts the knowledge of already formatted layers
+  const safeXAccessorLabelRenderer = (value: unknown): string =>
+    xAxisColumn && layersAlreadyFormatted[xAxisColumn.id]
+      ? (value as string)
+      : xAxisFormatter.convert(value);
```

Comment:
Something about this is not rendering any labels on the X axis when I'm in the "custom intervals" mode

Reply:
Do you have a screenshot or some steps to reproduce it?